lib/appsignal/phoenix.ex

```diff
@@ -17,30 +17,32 @@ if Appsignal.phoenix? do
 
     require Logger
 
-    alias Appsignal.{Transaction,TransactionRegistry}
+    @transaction Application.get_env(:appsignal, :appsignal_transaction, Appsignal.Transaction)
+    alias Appsignal.TransactionRegistry
     alias Appsignal.ErrorHandler
 
     @doc false
     defmacro __using__(_) do
       quote do
-        use Appsignal.Phoenix.Plug
-
         def call(conn, opts) do
           try do
             super(conn, opts)
           rescue
             e ->
               stacktrace = System.stacktrace
-            import Appsignal.Phoenix
-            case {Appsignal.TransactionRegistry.lookup(self()), extract_error_metadata(e, conn, stacktrace)} do
-              {nil, _} -> :skip
-              {_, nil} -> :skip
-              {transaction, {reason, message, stack, conn}} ->
-                submit_http_error(reason, message, stack, transaction, conn)
-            end
-            reraise e, stacktrace
+              import Appsignal.Phoenix
+              case {Appsignal.TransactionRegistry.lookup(self()), extract_error_metadata(e, conn, stacktrace)} do
+                {nil, _} -> :skip
+                {_, nil} -> :skip
+                {transaction, {reason, message, stack, conn}} ->
+                  submit_http_error(reason, message, stack, transaction, conn)
+              end
+              reraise e, stacktrace
           end
         end
+
+        defoverridable [call: 2]
```

Comment:
`Appsignal.Phoenix.call/2` overrides `YourPhoenixApp.Endpoint.call/2`, and is overridden by `Appsignal.Phoenix.Plug.call/2`. Each of the overrides calls `super/2`, ending us up with the following structure:

- Appsignal.Phoenix.Plug starts a transaction
  - Appsignal.Phoenix wraps the call in a try/catch block
    - call the original `YourPhoenixApp.Endpoint.call/2`
  - If an exception occurs in the try block, Appsignal.Phoenix sends an exception to AppSignal and removes the transaction from the registry
- Appsignal.Phoenix.Plug completes the transaction (if no exception was raised)